server/routes/github/github_signup.py

```
@@ -0,0 +1,598 @@
+
+
+def _mint_handoff(cur, user_id: str) -> str:
+    """Store a hashed one-time login token on the user row; return the raw token."""
+    token = secrets.token_urlsafe(32)
+    token_hash = hashlib.sha256(token.encode()).hexdigest()
+    cur.execute(
+        """UPDATE users
+              SET signup_handoff_hash = %s,
+                  signup_handoff_expires_at = %s
+            WHERE id = %s""",
+        (token_hash, datetime.now() + timedelta(seconds=HANDOFF_TTL_SEC), user_id),
```

Ruff (0.16.1)

[warning] `datetime.datetime.now()` called without a `tz` argument

(DTZ005)